morf-h2/src/main/java/org/alfasoftware/morf/jdbc/h2/H2Dialect.java

```diff
@@ -135,7 +134,17 @@ public H2Dialect() {
    */
   @Override
   public Collection<String> truncateTableStatements(Table table) {
-    return Arrays.asList("truncate table " + table.getName());
+    List<String> statements = new ArrayList<>();
+
+    StringBuilder truncateStatement = new StringBuilder();
+
+    truncateStatement.append("TRUNCATE TABLE ");
+
+    truncateStatement.append(schemaNamePrefix(table))
+                     .append(table.getName());
+
+    statements.add(truncateStatement.toString());
+    return statements;

```

Comment:
Avoid introducing new StringBuilders, only reuse them when they are there already.
Java compiler does StringBuilders already for us automatically, there is no added benefit.

https://niels.nu/blog/2016/stringbuilder.html Java is getting better :)